testsuites/sparql/src/main/java/org/eclipse/rdf4j/testsuite/sparql/tests/AggregateTest.java

```diff
@@ -432,48 +422,71 @@ public void testDistinctMax() {
 		try (TupleQueryResult result = conn.prepareTupleQuery(QueryLanguage.SPARQL, query).evaluate()) {
 			List<BindingSet> collect = QueryResults.asList(result);
 			int i = 0;
-			assertEquals(Values.literal(new BigDecimal("19.4786576482391284723864721567342354783275234")),
-					collect.get(i++).getValue("aggregate"));
-			assertEquals(Values.literal(23), collect.get(i++).getValue("aggregate"));
-			assertEquals(Values.literal(23), collect.get(i++).getValue("aggregate"));
-			assertEquals(Values.literal("2022-01-01T01:01:01.000000001Z", CoreDatatype.XSD.DATETIME),
-					collect.get(i++).getValue("aggregate"));
-			assertEquals(Values.literal("3"), collect.get(i++).getValue("aggregate"));
+			assertThat(collect.get(i++).getValue("aggregate"))
+					.isEqualTo(literal(new BigDecimal("19.4786576482391284723864721567342354783275234")));
+			assertThat(collect.get(i++).getValue("aggregate")).isEqualTo(literal(23));
+			assertThat(collect.get(i++).getValue("aggregate")).isEqualTo(literal(23));
+			assertThat(collect.get(i++).getValue("aggregate"))
+					.isEqualTo(literal("2022-01-01T01:01:01.000000001Z", CoreDatatype.XSD.DATETIME));
+			assertThat(collect.get(i++).getValue("aggregate")).isEqualTo(literal("3"));
 		}
+	}
+
+	/**
+	 * @see https://github.com/eclipse/rdf4j/issues/4290
+	 */
+	@Test
+	public void testCountOrderBy_ImplicitGroup() {
+		mixedDataForNumericAggregates();
 
+		String query = "select (count(*) as ?c) where { \n"
+				+ "	?s ?p ?o .\n"
+				+ "} \n"
+				+ "order by (?s)";
+
+		TupleQuery preparedQuery = conn.prepareTupleQuery(query);
+
+		List<BindingSet> result = QueryResults.asList(preparedQuery.evaluate());
+		assertThat(result).hasSize(1);
+
+		BindingSet bs = result.get(0);
+		assertThat(bs.size()).isEqualTo(1);
+		assertThat(getIntValue(bs.getValue("c"), 0)).isEqualTo(19);
 	}
```

Comment:
I got a bit carried away refactoring the tests, so the diff is a bit larger than ideal. This bit here is the actual new regression test.